RELEASE.md

```diff
@@ -0,0 +1,25 @@
+# Release Process
+
+At the moment, the release process is a manual process.
+
+## Prepare PR
+
+1. Update all links in `README.md` files from the previous version to the new version.
+2. Update the [examples](./examples) `main.tf` files to use the new version.
+3. Update the `mirror_image` in [docker-mirror](./modules/docker-mirror) to use the new version.
+4. Update the `executor_image` in [executors](./modules/executors) to use the new version.
+5. Open the PR
+6. At the moment, Buildkite will fail because the new version does not actually exist yet. So, a force merge is required.
+
+## Create Tag
+
+1. Create a tag matching the new version (e.g. `v4.2.0`).
```

Comment:
In between here we should add

```
## Merge the PR

1. Merge the PR from the GitHub UI
2. Check out `master` and pull the merged commit
```

(maybe I messed this up about 3 times at this point and had to force-push correct the tags.. duh)